### DISCUSSION TITLE
VS2022 show NotSignedIn but page show Device Activation Success

### DISCUSSION THREAD

**Discussion by MrTanXin** [community/community#14187] [Copilot Conversations]
## ENV:
Visual Studio Enterprise 2022 17.1.3
GitHub Copilot 1.10.0.3 (v1.10.0.3)
GitHub Copilot Agent v1.10.0

## Description:
1. When I Open Vs output pannel will show "A device activation code xxxx-xxxx has been copied to the clipboard." and pop-up "https://github.com/login/device"
2. I Enter code and Auth Success, page show "Congratulations, you're all set!",BUT VS not detected, nothing happend

**Comment by cool2528** [comment]
I also have the same problem wondering if anyone knows how to solve it?

**Reply by splendidhe** [reply]
the code had in your clipboard

**Reply by Flying932** [reply]
How to find my code,please?

**Reply by Flying932** [reply]
I don't know what device activation code is...

**Comment by kzu** [comment]
What happens if you click on the Copilot icon right next to the editor zoom level and try `Sign In` again from the menu?

**Reply by Johnson2580** [reply]
I also have the same problem，and when I click on the Copilot icon and click the sign in again ，it back to the first step。show "A device activation code xxxx-xxxx has been copied to the clipboard." and pop-up "https://github.com/login/device". Then I Enter code and Auth Success, page show "Congratulations, you're all set!",BUT VS shows “NotSignedIn”.

![image](https://user-images.githubusercontent.com/60962516/161882430-a56b1c46-ebaf-4268-b101-98a1bc53f132.png)

**Comment by cool2528** [comment]
@kzu There is no response, click sign in, continue to open the browser and enter the key, then no response.
![image](https://user-images.githubusercontent.com/42990446/161883671-11c381b0-186b-4ac1-8ec5-29bed6e035b7.png)

**Reply by Johnson2580** [reply]
哥们，我找到原因了，好像是VPN得关掉，我关掉后就登录成功了，但是自动提示还没搞出来，得点一下copilot里的trigger inline suggestion才能出来提示
![image](https://user-images.githubusercontent.com/60962516/161884902-91901f51-becf-4a40-ad76-0ece0fa6a10e.png)

**Reply by SolarianZ** [reply]
关了VPN后直接登录页面打不开了😂

**Reply by Fredrikwillion** [reply]
我没开vpn😂

**Reply by MrTanXin** [reply]
我关闭了 所有的VPN和代理 都没有效果 
尝试过切换浏览器也没效果 但是我的Jetbrains家族的软件是正常的

**Reply by cnfanhua** [reply]
同 Jetbrains家族的软件，vscode正常。vs2022不正常。太难了😢

**Reply by jfthuong** [reply]
> 同 Jetbrains家族的软件，vscode正常。vs2022不正常。太难了😢

Same for me (except for Jetbrains)

**Reply by sclbeta** [reply]
今天idea登不上了，不知道为什么，现象是一样的

**Reply by Flying932** [reply]
这个code是什么啊

**Comment by Fredrikwillion** [comment]
i also have this problem,please help me

**Comment by SmallTre** [comment]
Is there a solution to this problem?

**Comment by kzu** [comment]
The current (latest) version has a fix for this issue: it will properly clear a previously failed attempt to sign in with a different account and request brand new creds/device auth when you sign in.

Thanks for reporting the issue!

**Reply by GuilleZarle** [reply]
You don't have access to GitHub Copilot.

**Reply by jfthuong** [reply]
Latest version of Extension of VS 2022 or Extension?
I have  Extension version 1.14 and still has the same issue.

**Comment by RobotX5** [comment]
i also have this issue. how do i get the device id? it seems to break when it tries to open a web page. my guess it is a work security policy issue. can you please just add the device id to the output.

**Reply by kzu** [reply]
Device code is already shown in the output window. Thanks

**Reply by RobotX5** [reply]
the device code does not show for me in the output window.

**Reply by WspDev** [reply]
It's copied to the clipboard. Just paste it in.

**Comment by RobotX5** [comment]
Yes latest version of vs2022

**Reply by kzu** [reply]
Check the *Copilot* out window pane. It will be there for sure. If not, plz paste what you see in that pane (*not* the General pane) so we can help. 

Thanks

**Comment by cool2528** [comment]
Visual Studio 2022 Plugin Extension GitHub Copilot Activation Failure for Chinese Users Solution

 First turn off all your VPN proxies and modify your local dns master server cache to Google's `8.8.8.8` and use Aliyun's `223.5.5.5` for your backup dns server.
When you are done, press `WIN + R` on your keyboard and run `cmd`.
In the command line interface, type `ipconfig \flushdns` to refresh the dns cache
At this time, open your visual studio 2022 again and try to see if it works

**Comment by cool2528** [comment]
## Visual Studio 2022 插件扩展GitHub Copilot 中国用户激活失败解决办法

+ 首先关掉你的所有VPN代理 修改你的本地dns主服务器缓存 为Google的 `8.8.8.8` 备用dns服务器使用阿里云的`223.5.5.5`
+ 修改完毕以后按下键盘上的`WIN + R`  运行 `cmd`
+ 在命令行界面中输入 `ipconfig \flushdns` 刷新dns缓存
+ 这个时候再打开你的visual studio 2022 试试是否可以成功了

**Comment by RobotX5** [comment]
It started working for me on the 9th I think it may have been related to my vpn?

**Reply by WspDev** [reply]
Mine only works without the VPN.

**Comment by keytrap-x86** [comment]
To anyone having this problem, for me it was because of the proxy.
I switched to mobile sharing connection to activate the Copilot
